src/main/java/com/jmb/springfactory/dao/GenericMongoDaoImpl.java

```diff
@@ -0,0 +1,44 @@
+package com.jmb.springfactory.dao;
+
+import java.io.Serializable;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import org.springframework.data.mongodb.repository.MongoRepository;
+
+import com.jmb.springfactory.exceptions.PersistenceLayerException;
+import com.jmb.springfactory.model.entity.BaseEntity;
+
+public abstract class GenericMongoDaoImpl<T extends BaseEntity, ID extends Serializable> extends BaseDao implements 
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this generic name to match the regular expression '^[A-Z][0-9]?$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00119)